Highway.cpp

```diff
@@ -13,17 +14,41 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
+    if( auto* car = dynamic_cast<Car*>(v) )
+    {
+        car->closeWindows();
+    }
+
+    if( auto* semiTruck = dynamic_cast<SemiTruck*>(v) )
+    {
+        semiTruck->honk();
+    }
 
+    if( auto* motorcycle = dynamic_cast<Motorcycle*>(v) )
+    {
+        motorcycle->lanesplitAndRace(100);
+    }
```

Comment:
use `else if`
you're still checking if a Car is a SemiTruck or Motorcycle. 
- [ ] revise